pkg/cmd/secrets_edit.go

```diff
@@ -0,0 +1,199 @@
+	response, httpErr := http.GetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value)
+	if !httpErr.IsNil() {
+		utils.HandleError(httpErr.Unwrap(), httpErr.Message)
+	}
+
+	secrets, err := models.ParseSecrets(response)
+	if err != nil {
+		utils.HandleError(err, "Unable to parse API response")
+	}
+
+	return secrets
+}
+
+func saveSecrets(localConfig models.ScopedOptions, secrets map[string]interface{}) {
+	_, httpErr := http.SetSecrets(localConfig.APIHost.Value, utils.GetBool(localConfig.VerifyTLS.Value, true), localConfig.Token.Value, localConfig.EnclaveProject.Value, localConfig.EnclaveConfig.Value, secrets)
+	if !httpErr.IsNil() {
+		utils.HandleError(httpErr.Unwrap(), httpErr.Message)
+	}
+}
+
+func mapToEditFormat(selectedSecrets []string, secrets map[string]models.ComputedSecret) string {
+	buffer := header
+
+	for idx, val := range selectedSecrets {
+		buffer = buffer + val
+		buffer = buffer + "\n" + secrets[val].RawValue
+		if idx < len(selectedSecrets)-1 {
+			buffer = buffer + "\n\n\n"
+		}
+	}
+
+	return buffer
+}
+
+func parseEditFormat(buffer string) map[string]interface{} {
+	var firstRealLine string
+	scanner := bufio.NewScanner(strings.NewReader(buffer))
+	for scanner.Scan() {
+		text := scanner.Text()
+		if strings.HasPrefix(text, "#") || text == "" {
+			continue
+		}
+		firstRealLine = text
+		break
+	}
+	buffer = buffer[strings.Index(buffer, firstRealLine):]
+
+	secretsMap := map[string]interface{}{}
+	secrets := strings.SplitAfter(buffer, "\n\n\n")
+
+	for _, secret := range secrets {
+		secret = strings.Trim(secret, "\n")
+
+		secretArr := strings.SplitN(secret, "\n", 2)
+		name := secretArr[0]
+		value := ""
+		if len(secretArr) == 2 {
+			value = strings.TrimSuffix(secretArr[1], "\n")
+		}
+		secretsMap[name] = value
+	}
+
+	return secretsMap
+}
+
+func isEditableSecret(name string) bool {
+	if name == "DOPPLER_CONFIG" || name == "DOPPLER_ENVIRONMENT" || name == "DOPPLER_PROJECT" {
+		return false
+	}
+	return true
+}
+
+func selectSecrets(secrets map[string]models.ComputedSecret) []string {
+	var options []string
+
+	for _, val := range secrets {
+		if isEditableSecret(val.Name) {
+			options = append(options, val.Name)
+		}
+	}
+
+	sort.Strings(options)
+
+	return utils.MultiSelectPrompt("Select secrets:", options)
+}
+
+func EditInEditor(contents string) string {
+	tmpFd, err := utils.Memfile("fileToEdit", []byte(contents))
+	if err != nil {
+		utils.HandleError(err, "Unable to create file descriptor")
+	}
+
+	// filepath to our newly created in-memory file descriptor
+	tmpFilePath := fmt.Sprintf("/proc/self/fd/%d", tmpFd)
+
+	env := os.Environ()
+	editor := "vim"
+	for _, envVar := range env {
+		// key=value format
+		parts := strings.SplitN(envVar, "=", 2)
+		key := parts[0]
+		if strings.ToUpper(key) == "EDITOR" {
+			editor = parts[1]
+		}
+	}
```

Comment:
Since this surreptitiously relies on the environment, we may want to introduce out somewhat-standard `no-read-env` flag here.